evmjit/libevmjit/JIT.cpp

```diff
@@ -142,9 +147,37 @@ ExecFunc JITImpl::getExecFunc(h256 const& _codeHash) const
 
 void JITImpl::mapExecFunc(h256 _codeHash, ExecFunc _funcAddr)
 {
+	std::lock_guard<std::mutex> lock{x_codeMap};
 	m_codeMap.emplace(std::move(_codeHash), _funcAddr);
 }
 
+ExecFunc JITImpl::compile(byte const* _code, uint64_t _codeSize, h256 const& _codeHash)
+{
+	auto name = hash2str(_codeHash);
+	auto module = Cache::getObject(name);
+	if (!module)
+	{
+		// TODO: Listener support must be redesigned. These should be a feature of JITImpl
+		//listener->stateChanged(ExecState::Compilation);
+		assert(_code || !_codeSize); //TODO: Is it good idea to execute empty code?
```

Comment:
I would say that at least empty code should be supported. You can change it to an array of length 1 containing zero.